client/getipns.go

```diff
@@ -41,25 +41,37 @@ func (fp *Client) GetIPNSAsync(ctx context.Context, id []byte) (<-chan GetIPNSAs
 	ch1 := make(chan GetIPNSAsyncResult, 1)
 	go func() {
 		defer close(ch1)
-		if ctx.Err() != nil {
-			return
-		}
-		r0, ok := <-ch0
-		if !ok {
-			return
-		}
+
+		// TODO wrap in a for loop after fixing https://github.com/ipld/go-ipld-prime/issues/374
 		var r1 GetIPNSAsyncResult
-		if r0.Err != nil {
-			r1.Err = r0.Err
+		select {
+		case <-ctx.Done():
+			r1.Err = ctx.Err()
 			ch1 <- r1
-		} else if r0.Resp != nil {
+			return
+		case r0, ok := <-ch0:
+			if !ok {
+				return
+			}
+
+			if r0.Err != nil {
+				r1.Err = r0.Err
+				ch1 <- r1
+				return
```

Comment:
Errors returned through the channel do not mean termination of the stream. (I know this is not documented.) So, here remove the `return` statement in continue to the next iteration of the wrapping `for` loop.